cmd/config/operator/operator.go

```diff
@@ -0,0 +1,138 @@
+/*
+ * MinIO Cloud Storage, (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package operator
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/minio/minio/cmd/config"
+	"github.com/minio/minio/pkg/env"
+)
+
+// operator environment values
+const (
+	Endpoint  = "endpoint"
+	Namespace = "namespace"
+	Name      = "name"
+
+	EnvOperatorEndpoint = "MINIO_OPERATOR_ENDPOINT"
+)
+
+var (
+	globalRootCAs *x509.CertPool
+)
+
+// DefaultKVS - default KV settings for operator.
+var (
+	DefaultKVS = config.KVS{
+		config.KV{
+			Key:   Endpoint,
+			Value: "",
+		},
+	}
+)
+
+// Config - server operator config.
+type Config struct {
+	Enabled    bool   `json:"enabled"`
+	Endpoint   string `json:"endpoint"`
+	Username   string
+	Password   string
+	HTTPClient *http.Client
+}
+
+// RegisterGlobalCAs register the global root CAs
+func RegisterGlobalCAs(CAs *x509.CertPool) {
+	globalRootCAs = CAs
+}
+
+func parseEndpoint(endpoint string) (string, string, string, error) {
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	switch u.Scheme {
```

Comment:
You don't need to do any of this @nitisht all you are supposed to do is 

```
env.Get(EnvOperatorEndpoint, kvs.Get(endpoint))
```

This will automatically fetch the right endpoint by talking to remote operator if the `env` is `env:` you shouldn't be parsing `env` endpoint yourself.

Reply:
This doesn't work because, if the prefix is `env` env.Get  calls `getEnvValueFromHTTP` which attempts a `GET` request on the endpoint. While that is fine for `MINIO_ARGS`. `GET` is not supported for `MINIO_OPERATOR_ENDPOINT`.

Reply:
you should add that in operator @nitisht the endpoint is meant to get ENVs

Reply:
Why is a GET needed on bucketservice on operator?

Reply:
what does that achieve?

Reply:
Since you don't know the value of the env you should get it from the remote endpoint.